node/task.ts

```diff
@@ -243,41 +248,110 @@ export function loc(key: string, ...param: any[]): string {
 // Input Helpers
 //-----------------------------------------------------
 /**
- * Gets a variables value which is defined on the build definition or set at runtime.
+ * Gets a variable value that is defined on the build definition or set at runtime.
  * 
  * @param     name     name of the variable to get
  * @returns   string
  */
 export function getVariable(name: string): string {
     var varval;
-    if (vault) {
-        varval = vault.retrieveSecret('SECRET_' + name.replace(/\./g, '_').toUpperCase());
+
+    // get the metadata
+    var info: VariableInfo;
+    var key: string = getVariableKey(name);
+    if (variableInfo.hasOwnProperty(key)) {
+        info = variableInfo[key];
```

Comment:
I thought hasOwnProperty precaution is required due other folks extending base classes via prototype or something.